utilz/map_update.rb

```diff
@@ -0,0 +1,663 @@
+      opts.on("-p", "--projects", "Scrape Projects") { |v| @options[:projects] = v }
+      opts.on("-t", "--terms", "Scrape Terms") { |v| @options[:terms] = v }
+      opts.on("-u", "--termstxt", "Scrape Terms Text") { |v| @options[:terms_txt] = v }
+      opts.on("-T", "--writeterms", "Write Terms to MD") { |v| @options[:writeterms] = v }
+      opts.on("-I", "--termsindex", "Build Terms Index MD") { |v| @options[:termsindex] = v }
+      opts.on("-x", "--tidy DIRECTORY", "Tidy project YAML") { |v| @options[:tidy] = v }
+      opts.on("-X", "--drytidy", "DRY RUN Tidy project YAML (no files modified)") { |v| @options[:drytidy] = v }
+      opts.on("-g", "--geocode DIRECTORY", "Generate Longitude and Latitude automatically") { |v| @options[:geocode_projects] = v }
+      opts.on("-c", "--cache CACHE", "Write 'True' to activate Cache") { |v| @options[:cache_it] = v }
+      opts.on("-G", "--drygeo", "DRY RUN Generate Longitude and Latitude automatically") { |v| @options[:drygeo] = v }
+    end.parse!
+
+    # error raisers
+    unless @options[:tidy] or @options[:geocode_projects]
+      raise "ERROR! --input file not specified" if @options[:in_file].nil?
+      raise "ERROR! --infile does not exist" unless File.exist?(@options[:in_file])
+      raise "ERROR! --outdir is not a directory" unless File.directory?(@options[:out_dir])
+    end
+
+    # depending on calls made, call the functions
+    if @options[:projects]
+      scrape_projects_html
+    elsif @options[:terms]
+      scrape_terms_html
+    elsif @options[:terms_txt]
+      scrape_terms_txt
+    elsif @options[:writeterms]
+      write_terms_to_md
+    elsif @options[:termsindex]
+      build_terms_index
+    elsif @options[:tidy]
+      tidy_project_yml
+    elsif @options[:geocode_projects]
+      geocode_projects
+    else
+      p "nothing to do!"
+      puts optparse
+      exit
+    end
+  end
+
+  def self.geocode_projects
+    # ex:  ruby map_update.rb
+    p "DRY RUNNING GEOCODE PROCESS (good job!)" if @options[:drygeo]
+
+    @options[:geocode_projects] = "#{@options[:geocode_projects]}/" unless @options[:geocode_projects][-1] == '/'
+    p "Looking for MD files in #{@options[:geocode_projects]}"
+    all_filez = Dir.glob("#{@options[:geocode_projects]}**/*.md").select{ |e| File.file? e }
+    len = all_filez.length
+    idx = 0
+    all_filez.each do |file|
+      status_update(len:len, idx:idx)
+      project = read_md file: file
+
+      # find the place tag
+      if project[:yml]["place"] != ""
+        theplace = project[:yml]["place"]
+      else
+        theplace = ""
+      end
+
+      redis = Redis.new(host: "localhost")
+
+      # rest for a second so the requests don't come too fast
+      # for location iq, needs to be less than 2 requests per second
+      # for google, needs to be less than 50 requests per second
+      sleep(0.25)
+
+      @client = GooglePlaces::Client.new("AIzaSyCGucx-_XZJLn-Gd1cHme18vk6osePAE3w")
```

Comment:
# 🧐
hmm, yeah, you probably want to revoke this API key and setup another one and use an environment variable and not commit this api key to this public github repo. you can access env var values in ruby like `ENV["WHATEVER_ENV_VAR"]` and something like (there's a lot of options here but for eample:) `WHATEVER_ENV_VAR='someapikey' ruby some_script.rb -c True`

Reply:
ah HA okay, so THAT's how you do it - FYI as soon as I pushed this commit Google was like "wat" and immediately made me change the API key as I had uploaded it to a (as you mentioned) public repo. whoops.